synapse/storage/databases/main/events.py

```diff
@@ -439,6 +433,12 @@ def _persist_events_txn(
 
         self._store_event_txn(txn, events_and_contexts=events_and_contexts)
 
+        self._update_forward_extremities_txn(
+            txn,
+            new_forward_extremities=new_forward_extremities,
+            max_stream_order=max_stream_order,
+        )
```

Comment:
Any context behind this order of operations change?

Reply:
Just that we need to insert the event into the `events` table before we insert into `event_forward_extremities`.